source/features/search-releases.tsx

```diff
@@ -0,0 +1,155 @@
+import React from 'dom-chef';
+import select from 'select-dom';
+
+import features from '../libs/features';
+import {isReleasesOrTags} from '../libs/page-detect';
+import {getOwnerAndRepo} from '../libs/utils';
+import {octoface} from '../libs/icons';
+
+async function init() {
+	if (!isReleasesOrTags()) {
+		return false;
+	}
+
+	if (!isOnReleasePage()) {
+		return false;
+	}
+
+	addSelectMenu();
+	registerEventListener();
+}
+
+const isOnReleasePage = () => select.exists('.release-entry') || select.exists('.release');
+
+const getSelectedTagFromUrl = () => {
+	const result = location.href.match(/releases\/tag\/([^>/]*)/) || [];
+
+	return (result && result.length > 0) ? result[1] : null;
+};
+
+const addSelectMenu = () => {
+	const container = select('.subnav');
+	const {ownerName, repoName} = getOwnerAndRepo();
+	const selectedTag = getSelectedTagFromUrl();
+
+	container.append((
+		<div class="refined-github-search-releases float-right d-flex flex-shrink-0 flex-items-center mb-3">
+			<details class="details-reset details-overlay select-menu branch-select-menu">
+				<summary class="btn btn-sm select-menu-button css-truncate" data-hotkey="w" title="Filter releases" aria-haspopup="menu">
+					<i>Filter:</i>&nbsp;
+					<span class="css-truncate-target">{selectedTag || 'Releases'}</span>&nbsp;
+				</summary>
+				<details-menu
+					class="js-release-details select-menu-modal position-absolute"
+					src={`/${ownerName}/${repoName}/ref-list/master?source_action=disambiguate&source_controller=files`}
+					preload
+					role="menu"
+					style="z-index: 99;"
+				>
+					<include-fragment class="select-menu-loading-overlay anim-pulse">
+						{octoface()}
+					</include-fragment>
+				</details-menu>
+			</details>
+		</div>
+	));
+};
+
+const registerEventListener = () => {
+	const includeFragment = select('.js-release-details > include-fragment');
+
+	return includeFragment ? includeFragment.addEventListener('loadend', onFragmentLoaded) : null;
+};
+
+const onFragmentLoaded = () => {
+	adjustMenuStyles();
+	adjustTextInputStyles();
+	removeBranches();
+	adjustListStyles();
+	removeHeader();
+	removeTabs();
+	updateLinks();
```

Comment:
> These are all single-use functions. You can inline them all and avoid checking the elements in each.

The elements are being checked are different in each. I would say, I still check whether the element exists or not and if doesn't exist, log the error and return. What do you think?

> Usually, if the element isn’t present, we need the error to show so we can update the selector. `If false return` makes all errors silent.

Yeah, I agree with you here. Silently returning won't be helpful. 

Regarding the separation of function, I thought it would be better to keep them separate based on their functionality. But as they are very small, I will inline them.

Reply:
Why do you need the checks? To avoid errors? We need those errors to see when a features breaks. 

Also why do you want to log the error? Letting it throw already appears in the console and requires extra code. 

Most of those functions can be replaced by something like:

```js 
select('…').remove()
```

and 5 lines of that is more readable than 5 functions in 5 places amounting to at least 4 times the amount of lines each. It has no benefit and adds overhead for the reader. 

If a line does something incomprehensible, a comment can explain more than a function name (but in this case none of the function names help either)

Reply:
Yeah, You are right. Not avoiding the errors anymore. Also inlines all functions.